spec/models/mixins/event_mixin_spec.rb

```diff
@@ -41,19 +41,108 @@ def event_where_clause(assoc)
   end
 
   context "event_stream_filters" do
+    # Note
+    # 1) BottleneckEvent is not including event mixin, event_where_clause references missing resource_type, resource_id on event streams
+    # 2) EmsCluster#event_where_clause does an OR:  ems_cluster_id OR (host_id in ? OR dest_host_id IN ?) OR (vm_or_template in ? OR dest_vm_or_template_id in ?) (for now we just do ems_cluster_id in ems_event_filter)
```

Comment:
Yeah I think the idea was to show all cluster, host, and vm events, but that seems like overkill as a default. I think having cluster only is fine, and if you need more you can go to the hosts.  Perhaps in the future we have the cluster timeline have "Include Host events" or "Include VM events" checkboxes to allow for adding those in?

Reply:
Yes, that makes sense. We'll still probably have relevant uses for generating complex ANDs and ORs on the API and I couldn't get that to work properly before without support for grouping.  It's a todo thing.  I don't think it's terrible as a first time replacement but want to figure out something to replace event_where_clause.

As mentioned elsewhere, we don't even present some of the event_where_clause classes in the UI and others we don't really implement policy / miq events properly for them.  In many cases, the policy events are the same as ems events, which is wrong.

Reply:
Honestly, I want to discard any junk we don't need and simplify it.  I like the idea of making it simpler and perhaps have the UI split up OR conditions into separate AND API calls.  Of cousre, the backend could just return an array of hash conditions instead of just a single hash right now.

Reply:
Perhaps we can also pass an MiqExpression as part of the payload if we need the complexity 🤔